src/support/mixins/misc.scss

```diff
@@ -24,3 +24,12 @@
     background-color: $border;
   }
 }
+
+// global focus styles
+
+@mixin focusOutline {
+  position: relative;
+  z-index: 1;
+  outline: 2px solid var(--color-accent-fg);
```

Comment:
What do you think about using a dedicated custom property for outline-color? This would enable background color utilities to set —focus-outline-color when necessary to ensure sufficient contrast between the focus outline and the background color.

Reply:
@simurai @auareyou any thoughts here?

Reply:
Hmm.. yeah, that was still an open question in @vdepizzol's proposal: https://github.com/github/primer/issues/410

![image](https://user-images.githubusercontent.com/378023/142334648-7d217bdf-ff8f-48d3-8e2e-e952dbb01a24.png)

> @jdanyow What do you think about using a dedicated custom property for outline-color? This would enable background color utilities to set —focus-outline-color when necessary to ensure sufficient contrast between the focus outline and the background color.

That's a good point. There are areas like the [Checks](https://github.com/primer/css/runs/4245906701?check_suite_focus=true) or the main Header that have "hard coded" dark backgrounds where `--color-accent-fg` might not have enough contrast:

![Screen Shot 2021-11-18 at 10 53 13](https://user-images.githubusercontent.com/378023/142337260-0e94403e-5360-4a82-9fff-d011b9cd577d.png)

**Short term** we _could_ add a 2nd white (`--color-canvas-default`) outline to improve contrast for these cases, but that doesn't look that great. It's what currently the browser default (Chrome) does:

![Screen Shot 2021-11-18 at 10 53 22](https://user-images.githubusercontent.com/378023/142337277-310f5f4e-0259-4421-b388-d17aa67eb893.png)

Or yes, add a new `--color-primer-focus-outline` to Primer Primitives as well as a `--color-checks-focus-outline` that is optimized to be used on a dark background also in light mode. Then somewhere on dotcom we could override the `--color-primer-focus-outline` variable like:

```scss
.CheckRun {
  --color-primer-focus-outline: var(--color-checks-focus-outline);
}
```

**Long term** we still have to look into supporting "[nested themes](https://github.com/github/design-infrastructure/discussions/1394)".

Reply:
Do we have a resolution here? It sounds to me like we should be focusing on fixing areas that have hard-coded theme values 🤔 given that this is global CSS, I feel like we should try and ensure it works for the majority of cases while not adding too much complexity for edge cases.